src/solvers/flattening/boolbv_mult.cpp

```diff
@@ -12,53 +12,40 @@ Author: Daniel Kroening, kroening@kroening.com
 
 bvt boolbvt::convert_mult(const exprt &expr)
 {
-  std::size_t width=boolbv_width(expr.type());
-
+  const std::size_t width = boolbv_width(expr.type());
   if(width==0)
     return conversion_failed(expr);
-
   bvt bv;
   bv.resize(width);
 
   const exprt::operandst &operands=expr.operands();
-  if(operands.empty())
-    throw "mult without operands";
-
+  INVARIANT(!operands.empty(), "mult must have operands");
   const exprt &op0=expr.op0();
-
-  bool no_overflow=expr.id()=="no-overflow-mult";
+  const bool no_overflow = expr.id() == "no-overflow-mult";
```

Comment:
Would be good to add an irepid for this.

Reply:
Actually it does not seem any `exprt` gets this id anywhere. So it may be better to just remove it. And same thing with the other `no-overflow`s

Reply:
@kroening Is there code elsewhere that generates `no-overflow` variants of the arithmetic expressions?

Reply:
If there isn't then there should be; it is useful to know whether overflow matters or not / can be ignored.  Especially for multiplication.